sdk/core/core-auth/src/azureKeyCredential.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Represents a credential defined by a static API key.
+ */
+export interface KeyCredential {
+  /**
+   * The value of the API key represented as a string
+   */
+  key: string;
+}
+
+/**
+ * A static-key-based credential that supports updating
+ * the underlying key value.
+ */
+export class AzureKeyCredential implements KeyCredential {
+  /**
+   * The value of the key to be used in authentication
+   */
+  public key: string;
```

Comment:
Could we make this property private and just have a public getter? This way we don't have two ways of updating the key. It doesn't really matter that much for this particular credential since it's just one value that makes up the key, but other credential types like this have multiple fields and so the purpose of the update method is to swap them in a thread safe way. I'd like to make this consistent with that pattern so credential types throughout our sdk provided a similar experience.

Reply:
@schaabs I just put up some changes to the design that account for this. There's now a getter that provides access to the underlying key, and this is represented in the `KeyCredential` interface by marking it `readonly`.